### ISSUE TITLE
Add synchronization around I2C

### ISSUE THREAD

**Issue by savageautomate** [Pi4J/pi4j-v1#46]
REF:  https://groups.google.com/forum/#!topic/pi4j/CQoVN1eU01Q

FROM: rlsutton1 

I have a number of I2C devices attached to my PI (Gyro, ADC + Adafruit PWM), and am polling/writting them at up to 100hz. I found that the I2C bus was failing.

would it be possible to add some synchronization to the pi4j libs around I2C? 

In the mean time I added a synchronized wrapper to the I2C Device which solved my problems, thought I'd share the wrapper class here. (see attached).

usage looks like this...

```
    bus = I2CFactory.getInstance(busNumber);

    // create I2C device instance
    device = new SynchronizedDeviceWrapper(bus.getDevice(address));
```

**Comment by JanJansen47** [comment]
Thank for for this issue. Maybe I have the same  nasty problem which occurs add-hoc. 
(I'm using custom build i2c based I/O but used the Pi4J classes up to now)

**Comment by savageautomate** [comment]
Added locks from SynchronizedDeviceWrapper into I2CDeviceImpl.
Available in latest 1.0-SNAPSHOT build
